libsolidity/analysis/DocStringAnalyser.cpp

```diff
@@ -32,6 +32,49 @@ using namespace solidity;
 using namespace solidity::langutil;
 using namespace solidity::frontend;
 
+namespace {
+
+void copyMissingTags(StructurallyDocumentedAnnotation& _target, CallableDeclarationAnnotation const& _source)
+{
+	auto& sourceDoc = dynamic_cast<StructurallyDocumentedAnnotation const&>(_source);
+
+	for (auto iterator = sourceDoc.docTags.cbegin(); iterator != sourceDoc.docTags.cend();)
+		if (_target.docTags.count(iterator->first) || iterator->first == "inheritdoc")
+			iterator++;
+		else
+			for (auto nextTag = sourceDoc.docTags.upper_bound(iterator->first); iterator != nextTag; iterator++)
+				_target.docTags.emplace(iterator->first, iterator->second);
```

Comment:
I may not have understood this properly, but would the following work as a replacement?
```cpp
for (auto [key, value]: sourceDoc.docTags)
	if (!_target.docTags.count(key) && key != "inheritdoc")
		_target.docTags.emplace(key, value);
```

Reply:
Unfortunately not, because it is a multimap. I there is a multi-value entry in the source, after the first insert in the target, it would no longer insert the other parts

Reply:
Ah, I missed that.

Reply:
How about we make a set of tags present in target first? Then we only check for keys from this set?

Reply:
That would work, yes

Reply:
I am not thrilled about the extra memory usage, but if it makes it easier to read we can do it that way

Reply:
I think the memory usage is minimal, since the set can contain at most 6 elements.

Reply:
I have to admit, the result looks a lot nicer.